php/libraries/ExternalIDGenerator.php

```diff
@@ -27,9 +27,7 @@ class ExternalIDGenerator extends SiteIDGenerator
     /**
      * Generates ExternalIDs.
      *
-     * @param ?string $prefix The site prefix to prepend to the ID value.
-     *
-     * @return void
+     * @param string|null $prefix The site prefix to prepend to the ID value.
```

Comment:
`?string` is a valid annotation, just FYI

Reply:
@johnsaigle thanks, you want me to revert those? 
The string|null is more descriptive to me from being someone who isn't a php expert. Personally I had to google the ?string when we introduced type hinting to know what was going on. Although I can understand people would prefer the shorthand annotation. I can revert if you want.

Reply:
No it's not worth the effort of going back and changing it. I like it in the sense that it matches type hinting for params and return types but it is initially confusing. Just wanted to let you know that it's not incorrect.